src/container/builder_impl.rs

```diff
@@ -124,12 +124,12 @@ fn container_init(
     notify_name: PathBuf,
     child: &mut child::ChildProcess,
 ) -> Result<()> {
-    let linux = &spec.linux;
+    let linux = &spec.linux.as_ref().context("no linux spec")?;
     let namespaces: Namespaces = linux.namespaces.clone().into();
     // need to create the notify socket before we pivot root, since the unix
     // domain socket used here is outside of the rootfs of container
     let mut notify_socket: NotifyListener = NotifyListener::new(&notify_name)?;
-    let proc = &spec.process;
+    let proc = &spec.process.as_ref().context("no process")?;
```

Comment:
Can you change this to "no process spec" here and everywhere else? My first reaction to "no process" in an error message would be to assume that the container process has exited, not that some part of the spec is missing.

Reply:
Yep, changed as suggested.